### PR TITLE
Implement full TEE with sgx

### PR DESCRIPTION
**Pull request by zguesmi**

### CODE REVIEW

File: src/main/java/com/iexec/common/replicate/ReplicateStatus.java [hunk 1]
```diff
@@ -20,6 +20,9 @@
     COMPUTED,
     COMPUTE_FAILED,
     CAN_CONTRIBUTE,
+    CANT_CONTRIBUTE_SINCE_DETERMINISM_HASH_NOT_FOUND,
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/replicate/ReplicateStatus.java:23]: We should update getUncompletableStatuses() and co

File: src/main/java/com/iexec/common/utils/SignatureUtils.java [hunk 2]
```diff
@@ -39,6 +40,27 @@ public static boolean doesSignatureMatchesAddress(byte[] signatureR,
         return false;
     }
 
+    public static boolean isSignatureValid(String message, Signature sign, String signerAddress) {
+        return isSignatureValid(message.getBytes(), sign, signerAddress);
+    }
+
+    public static boolean isSignatureValid(byte[] message, Signature sign, String signerAddress) {
```
Comment (Ugo) [inline src/main/java/com/iexec/common/utils/SignatureUtils.java:43]: This is an important method here, good unit tests are mandatory.

**Review by Ugo — CHANGES_REQUESTED** [review]

File: src/main/java/com/iexec/common/utils/SignatureUtils.java (hunk 2 above)
Comment (zguesmi) [reply src/main/java/com/iexec/common/utils/SignatureUtils.java:43]: Sure yes! thank you for the note.

File: src/main/java/com/iexec/common/replicate/ReplicateStatus.java (hunk 1 above)
Comment (zguesmi) [reply src/main/java/com/iexec/common/replicate/ReplicateStatus.java:23]: Done, thank you!

**Review by james-toussaint — APPROVED** [review]

**Review by Ugo — APPROVED** [review]

**Comment by Ugo** [comment]
Thank you Zied!

**Pull request merged**